src/ol_infrastructure/applications/keycloak/__main__.py

```diff
@@ -0,0 +1,418 @@
+"""The complete state needed to provision Keycloak running on Docker.
+
+"""
+import base64
+import json
+import textwrap
+from itertools import chain
+from pathlib import Path
+
+import pulumi_consul as consul
+import pulumi_vault as vault
+import yaml
+from pulumi import Config, Output, ResourceOptions, StackReference, export
+from pulumi_aws import ec2, get_caller_identity, iam, route53
+from pulumi_consul import Node, Service, ServiceCheckArgs
+
+from bridge.lib.magic_numbers import (
+    AWS_RDS_DEFAULT_DATABASE_CAPACITY,
+    DEFAULT_HTTPS_PORT,
+    DEFAULT_POSTGRES_PORT,
+)
+from bridge.secrets.sops import read_yaml_secrets
+from ol_infrastructure.components.aws.auto_scale_group import (
+    BlockDeviceMapping,
+    OLAutoScaleGroupConfig,
+    OLAutoScaling,
+    OLLaunchTemplateConfig,
+    OLLoadBalancerConfig,
+    OLTargetGroupConfig,
+    TagSpecification,
+)
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLPostgresDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultPostgresDatabaseConfig,
+)
+from ol_infrastructure.lib.aws.ec2_helper import InstanceTypes, default_egress_args
+from ol_infrastructure.lib.consul import consul_key_helper, get_consul_provider
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+# Configuration items and initialziations
+if Config("vault_server").get("env_namespace"):
```

Comment:
We can remove these conditionals. That was largely a "feature flag" while we got this automatic provider setup working and migrated over.